test/hotspot/gtest/runtime/test_os.cpp

```diff
@@ -494,11 +495,32 @@ TEST_VM(os, release_one_mapping_multi_commits) {
   PRINT_MAPPINGS("D");
 }
 
-TEST_VM(os, show_mappings_1) {
-  // Display an arbitrary large address range. Make this works, does not hang, etc.
-  char dummy[16 * K]; // silent truncation is fine, we don't care.
-  stringStream ss(dummy, sizeof(dummy));
-  os::print_memory_mappings((char*)0x1000, LP64_ONLY(1024) NOT_LP64(3) * G, &ss);
+static void test_show_mappings(address start, size_t size) {
+  // Note: should this overflow, thats okay. stream will silently truncate. Does not matter for the test.
+  const size_t buflen = 4 * M;
+  char* buf = NEW_C_HEAP_ARRAY(char, buflen, mtInternal);
+  buf[0] = '\0';
+  stringStream ss(buf, buflen);
+  if (start != NULL) {
+    os::print_memory_mappings((char*)start, size, &ss);
+  } else {
+    os::print_memory_mappings(&ss); // prints full address space
+  }
+  // Still an empty implementation on MacOS and AIX
+#if defined(LINUX) || defined(_WIN32)
+  EXPECT_NE(buf[0], '\0');
+#endif
+  // buf[buflen - 1] = '\0';
+  // tty->print_raw(buf);
+  FREE_C_HEAP_ARRAY(char, buf);
+}
+
+TEST_VM(os, show_mappings_small_range) {
+  test_show_mappings((address)0x100000, 2 * G);
+}
+
+TEST_VM(os, show_mappings_full_range) {
+  test_show_mappings(NULL, 0);
```

Comment:
According to hotspot-style.md nullptr should be preferred.

Reply:
Sigh. 

I have an unreasonable dislike against nullptr and the churn that causes (NULL is nullptr anyway post C++11). But I'll be a good boy and change all those :)